pkg/auth/userregistry/identitymapper/provision.go

```diff
@@ -0,0 +1,153 @@
+package identitymapper
+
+import (
+	"errors"
+	"fmt"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	kerrs "github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
+	kuser "github.com/GoogleCloudPlatform/kubernetes/pkg/auth/user"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+
+	authapi "github.com/openshift/origin/pkg/auth/api"
+	"github.com/openshift/origin/pkg/user"
+	userapi "github.com/openshift/origin/pkg/user/api"
+	identityregistry "github.com/openshift/origin/pkg/user/registry/identity"
+	userregistry "github.com/openshift/origin/pkg/user/registry/user"
+)
+
+type UserNameGenerator func(base string, sequence int) (string, error)
+
+var (
+	MaxGenerateAttempts = 100
+	DefaultGenerator    = UserNameGenerator(func(base string, sequence int) (string, error) {
+		switch {
+		case sequence == 0:
+			return base, nil
+		case sequence > MaxGenerateAttempts:
```

Comment:
You should not be indirectly controlling the later for loop from this location.  The loop itself should enforce limits on the number of iteration attempts.  Doing it here makes it difficult to read.